tests/server/test_databaseservice.py

```diff
@@ -40,3 +44,56 @@ async def _wait_until_expire_is_finished():
     # Assert cleanup
     result = await data.AgentProcess.get_list()
     assert len(result) == 1
+
+
+@pytest.fixture(scope="function")
+async def set_pool_size_to_one():
+    """
+    Sets the database pool's max and min size to 1 and resets their prior values upon exiting.
+    """
+    save_min_size = opt.db_connection_pool_min_size.get()
+    save_max_size = opt.db_connection_pool_max_size.get()
+
+    opt.db_connection_pool_min_size.set("1")
+    opt.db_connection_pool_max_size.set("1")
+
+    yield
+
+    opt.db_connection_pool_min_size.set(str(save_min_size))
+    opt.db_connection_pool_max_size.set(str(save_max_size))
+
+
+@pytest.mark.slowtest
+async def test_pool_exhaustion_watcher(set_pool_size_to_one, server, caplog):
+    """
+    Test the basic functionalities of the ExhaustedPoolWatcher class
+    """
+    with caplog.at_level(logging.WARNING, logging.getLogger("inmanta.server.services.databaseservice").name):
+        database_slice = server.get_slice(databaseservice.SLICE_DATABASE)
+
+        assert database_slice._db_pool_watcher._exhausted_pool_events_count == 0
```

Comment:
This assertion is not always true. This test case runs an Inmanta server with a database pool size of 1. The server executes several background task, that might need a database connection. So at this point there might already be a pool exhaustion event.